patroni/utils.py

```diff
@@ -195,7 +195,7 @@ def _sleep(interval):
 
 
 def is_valid_pg_version(version):
-    return re.match(r'[1-9][0-9]?(\.(0|([1-9][0-9]?))){2}$', version)
+    return re.match(r'[1-9][0-9]?(\.(0|([1-9][0-9]?))){1,2}$', version)
```

Comment:
This is still wrong, although there are little negative consequences of that wrongness. The regex accepts 9.3 and 9.5 as a valid PostgreSQL versions, although it passes it to the postgres_version_to_int that expects major.minor versions (although postgres_version_to_int also doesn't know 9.3 is not a proper syntax for the major.minor version before 10, so it gets passed). From the user perspective, it seems that we accept just the major version, and it actually works fine with 9.x major versions, but then we don't accept 10 or 11, perfectly valid major versions.

What we should do is to accept only full version, i.e. major.minor for 10 and above or 9.major.minor.

Also, the code in the error message on the line 473 still complains about first.major.minor. It should say something like 'PostgreSQL version should be in the 9.major.minor below version 10 or major.minor otherwise.' (the example on the line 464 should also add the new version format, e.g. 9.5.2 or 10.1).

The test in test_ctl.py:240 actually caught the issue correctly, but you've modified it.